src/biotite/database/uniprot/download.py

```diff
@@ -0,0 +1,190 @@
+__author__ = "Maximilian Greil"
+__all__ = ["_get_database_name", "fetch", "_assert_valid_file"]
+
+from os.path import isdir, isfile, join, getsize
+import os
+import io
+import requests
+from ..error import RequestError
+
+fetch_url = "https://www.uniprot.org/"
+
+
+def _get_database_name(uid):
+    """
+    Get the correct UniProt database from the ID of the file to be downloaded.
+
+    Parameters
+    ----------
+    uid: str
+        ID of the file to be downloaded.
+
+    Returns
+    -------
+    name : str
+        E-utility UniProt database name.
+    """
+    if uid[:3] == "UPI":
+        return "uniparc"
+    elif uid[:9] == "UniRef90_":
+        return "uniref"
+    return "uniprot"
+
+
+def fetch(uids, format, target_path=None,
+          overwrite=False, verbose=False):
+    """
+    Download structure files (or sequence files) from the UniProt in
+    various formats.
+
+    Available databases are UniProtKB, UniRef and UniParc
+
+    This function requires an internet connection.
+
+    Parameters
+    ----------
+    uids : str or iterable object of str
+        A single ID or a list of IDs of the file(s)
+        to be downloaded.
+    format : {'fasta', 'gff', 'txt', 'xml', 'rdf', 'tab'}
+        The format of the files to be downloaded.
+    target_path : str, optional
+        The target directory of the downloaded files.
+        By default, the file content is stored in a file-like object
+        (`StringIO` or `BytesIO`, respectively).
+    overwrite : bool, optional
+        If true, existing files will be overwritten. Otherwise the
+        respective file will only be downloaded if the file does not
+        exist yet in the specified target directory or if the file is
+        empty. (Default: False)
+    verbose: bool, optional
+        If true, the function will output the download progress.
+        (Default: False)
+
+    Returns
+    -------
+    files : str or StringIO or BytesIO or list of (str or StringIO or BytesIO)
+        The file path(s) to the downloaded files.
+        If a single string (a single ID) was given in `pdb_ids`,
+        a single string is returned. If a list (or other iterable
+        object) was given, a list of strings is returned.
+        If no `target_path` was given, the file contents are stored in
+        either `StringIO` or `BytesIO` objects.
+
+    Warnings
+    --------
+    Even if you give valid input to this function, in rare cases the
+    database might return no or malformed data to you.
+    In these cases the request should be retried.
+    When the issue occurs repeatedly, the error is probably in your
+    input.
+
+    Examples
+    --------
+
+    >>> import os.path
+    >>> file = fetch("P12345", "fasta", path_to_directory)
+    >>> print(os.path.basename(file))
+    P12345.fasta
+    >>> files = fetch(["P12345", "Q8K9I1"], "fasta", path_to_directory)
+    >>> print([os.path.basename(file) for file in files])
+    ['P12345.fasta', 'Q8K9I1.fasta']
```

Comment:
Could you also add the `biotite.database.uniprot` subpackage as test case to `tests/test_doctest.py::test_doctest()`? This way the doctest code output is tested by *pytest*.